src/server/components/lists/find/middleware/loadQueryParametersIntoSession.ts

```diff
@@ -0,0 +1,91 @@
+import type { NextFunction, Request, Response } from "express";
+import { sanitisePracticeAreas } from "server/components/lists/find/helpers/sanitisePracticeAreas";
+import { sanitiseServices } from "server/components/lists/find/helpers/sanitiseServices";
+import { sanitiseLanguages } from "server/components/lists/find/helpers/sanitiseLanguages";
+import { sanitiseTranslationTypes } from "server/components/lists/find/helpers/sanitiseTranslationTypes";
+import { sanitiseInterpretationTypes } from "server/components/lists/find/helpers/sanitiseInterpretationTypes";
+import { languageCodesToReadable } from "server/components/lists/find/helpers/languageCodeToReadable";
+import type { ParsedQs } from "qs";
+
+export function loadQueryParametersIntoSession(req: Request, res: Response, next: NextFunction) {
+  const { serviceType, region, country } = req.query;
+  req.session.answers = {
+    region: region as string,
+    country: queryParameterValueAsString(country),
+    disclaimer: true,
+  };
+
+  if (serviceType === "funeralDirectors") {
+    loadFuneralDirectorsQueryParameters(req, res, next);
+    return;
+  }
+
+  if (serviceType === "translatorsInterpreters") {
+    loadTranslatorsInterpretersQueryParameters(req, res, next);
+    return;
+  }
+
+  if (serviceType === "lawyers") {
+    loadLawyersQueryParameters(req, res, next);
+    return;
+  }
+  next();
+}
+function loadFuneralDirectorsQueryParameters(req: Request, res: Response, next: NextFunction) {
+  req.session.answers = {
+    ...req.session.answers,
+    practiceAreas: sanitisePracticeAreas(req.query["practice-area"] as string),
+    repatriation: req.query.repatriation === "yes",
```

Comment:
if req.query.repatriation is not specified because you've just submitted the region page, for example, will this default to false?

Reply:
Not sure what you mean RE: submission. For submissions now, you will find the answer to repat on `req.body`.

`loadFuneralDirectorsQueryParameters` - as well as matching middleware for the other service types are to help convert any user which has old style urls (answers in the query parameters). 

In the event that links have been distributed, or bookmarked with old URLs, this ensures that the users sees the same thing if they revisit this link. 

If you have saved this URL - it should not have been possible to have "not specified" or have this question unanswered.

Reply:
ok that makes more sense now.